pkg/pillar/evetpm/tpm.go

```diff
@@ -625,11 +648,31 @@ func SealDiskKey(key []byte, pcrSel tpm2.PCRSelection) error {
 		return fmt.Errorf("NVWrite %v failed: %v", TpmSealedDiskPubHdl, err)
 	}
 
-	// save a snapshot of PCR values
+	// save a snapshot of current PCR values
```

Comment:
My understanding is that one needs to match the PCRs with the event log, but we have one file with the saved PCRs and 4 files with saved event logs. That seems like a mismatch and I don't understand exactly why one wouldn't want the saved PCRs which go with the saved event logs.

Reply:
the mismatched PCR error goes to logs and eventually in the cloud, so I kept it there if for some reason we lost the locally saved event logs, still have something to work with. If you find it redundant, I will remove the `saveDiskKeySealingPCRs`.

Reply:
Why not have the old PCRs and their associated event log (the one which sealed and worked to unseal), and the new PCRs and associated event log which fails to unseal? Why do you need more event logs and want less PCRs?

Reply:
1. We save old PCRs (seal worked), but there is no need to save new PCRs (unseal fail), because we read it from TPM, do a diff with the old PCRs and just report the mismatch in the error logs, so the information is preserved. Then when backup key is retrieved from the cloud controller, key gets sealed under the current PCRs (which is equal to new PCRs [unseal fail]) and we update the old PCRs (seal worked) file with these current values so it serves as the "seal worked" snapshot for future error detection.

2. We save the old event log (seal worked), and when there is an error, we also save new event log (unseal failed), the reason we are not doing the same thing as 1 (diffing old and new and just reporting diff result in error logs) is that diffing results in hard to understand data with lots of extra helpful info missing. At this point we have two event logs saved, the old and the new which can be retrieved from device for further diagnosis. But often at this stage EVE is connected to the controller and gets the backup key. Key is again sealed under the current PCRs (and the associated event log) and we update the old event log (seal worked) with the current one so it serves as the "seal worked" snapshot of event log for future error detection. The problem is that current event log is same as "unseal fail" event log, and by saving it in the old event log, we end up with two files with same contents. So to preserve the debug information, we rename the old and new event logs to something like old.backup and new.backup and save the current "unseal fail" as new event log.

Reply:
OK. But it would be great to provide an example of how to use this to debug unseal failures in some appropriate markdown file in EVE. Can be done as a separate PR.

Reply:
added to debug.md https://github.com/lf-edge/eve/pull/3405